### ISSUE TITLE
Server-side Fabric mod/plugin support

### ISSUE THREAD

**Issue by Titaniumtown** [cuberite/cuberite#4752]
Ok, so I know that non-lua based plugins aren't the best solution to implement in cuberite. But wasn't the reason why Bukkit support was completely off the table was because of licensing? Fabric has the same license as cuberite, so there shouldn't be a problem there right? If this is legally possible, it would allow people to be able to use stuff like https://github.com/ViaVersion/ViaFabric instead of having a waterfall proxy and viaversion. 

I know this would be a long term endevour, but my question is that if it's legally possible.

Thoughts?

**Comment by QuImUfu** [comment]
**This is just my two cents and may be wrong in one way or another^^:**  

Implementing the fabric API would probably be no huge problem, but it wouldn't make sense (at all). Fabric has a very minimal API and is designed to allow madders to implement their mods directly atop vanilla. This means all mods access (some only) vanilla functions/classes.  

Implementing a proxy for all vanilla classes and methods is as legally problematic as implementing the bukkit API (the legality of both depends on whether API's are copyrightable) and would require tons of work and clean-room reverse engineering that internal API (**that changes with each version**). Additionally mods use mixins to directly change minecraft code, which would be (nigh) impossible to emulate.  
Long story short:
Nigher technically nor legally feasible.

**Comment by DaMatrix** [comment]
bukkit and fabric and forge and sponge and all the other APIs are java, whereas cuberite is c++. adding bukkit support would require bundling the entire JVM into the server, as well as redesigning a lot of things to make it work well with bukkit.

**Comment by Titaniumtown** [comment]
I remember that there was talks about how legally bukkit support could be implemented, because of licensing and stuff. So i'm asking from a legal perspective.

**Comment by DaMatrix** [comment]
there wouldn't be any legal issues from bukkit (see projects like https://github.com/GlowstoneMC/Glowstone/), legal issues would probably arise from use of the JVM in that way. in any rate the implementation would be insanely complex and provide relatively little benefit compared to the difficulty of implementing it. fabric would be completely impossible, as it relies on mods transforming base minecraft code at runtime, of which cuberite has none.
